File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Windows.Forms;

namespace FilesPreviewGenerator
{
    public partial class Form1 : Form
    {
        List<FileInfo> files;

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            InitData();
            gridControl1.DataSource = files;

            InitViewSettings();
            winExplorerView1.GetThumbnailImage += WinExplorerView_GetThumbnailImage;
        }
   
        private void InitData()
        {
            files = new List<FileInfo>();
            string[] fileNames = Directory.GetFiles("Documents\\");
            foreach (string item in fileNames)
            {
                files.Add(new FileInfo(item, Path.GetFileName(item), Path.GetExtension(item)));
            }

        }
        private void InitViewSettings()
        {
            winExplorerView1.OptionsImageLoad.CacheThumbnails = true;
            winExplorerView1.OptionsImageLoad.LoadThumbnailImagesFromDataSource = false;
            winExplorerView1.OptionsImageLoad.AsyncLoad = true;
            winExplorerView1.OptionsView.Style = DevExpress.XtraGrid.Views.WinExplorer.WinExplorerViewStyle.ExtraLarge;
        }

        private void WinExplorerView_GetThumbnailImage(object sender, DevExpress.Utils.ThumbnailImageEventArgs e)
        {
            string filePath = files[e.DataSourceIndex].Path;
            string documentFormat = files[e.DataSourceIndex].DocumentFormat;

            switch (documentFormat)
            {
                case ".pdf":
                    {
                        int largestEdgeLength = Math.Max(e.DesiredThumbnailSize.Width, e.DesiredThumbnailSize.Height);
                        e.ThumbnailImage = ImageExporterHelper.GenerateImageFromPDF(filePath, largestEdgeLength);
                    }
                    break;
                case ".xlsx":
                        e.ThumbnailImage = ImageExporterHelper.GenerateImageFromExcel(filePath);
                    break;
                case ".docx":
                        e.ThumbnailImage = ImageExporterHelper.GenerateImageFromWord(filePath);
                    break;
            }

        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Windows.Forms

Namespace FilesPreviewGenerator

    Public Partial Class Form1
        Inherits Form

        Private files As List(Of FileInfo)

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            InitData()
            gridControl1.DataSource = files
            InitViewSettings()
            AddHandler winExplorerView1.GetThumbnailImage, AddressOf WinExplorerView_GetThumbnailImage
        End Sub

        Private Sub InitData()
            files = New List(Of FileInfo)()
            Dim fileNames As String() = Directory.GetFiles("Documents\")
            For Each item As String In fileNames
                files.Add(New FileInfo(item, Path.GetFileName(item), Path.GetExtension(item)))
            Next
        End Sub

        Private Sub InitViewSettings()
            winExplorerView1.OptionsImageLoad.CacheThumbnails = True
            winExplorerView1.OptionsImageLoad.LoadThumbnailImagesFromDataSource = False
            winExplorerView1.OptionsImageLoad.AsyncLoad = True
            winExplorerView1.OptionsView.Style = DevExpress.XtraGrid.Views.WinExplorer.WinExplorerViewStyle.ExtraLarge
        End Sub

        Private Sub WinExplorerView_GetThumbnailImage(ByVal sender As Object, ByVal e As DevExpress.Utils.ThumbnailImageEventArgs)
            Dim filePath As String = files(e.DataSourceIndex).Path
            Dim documentFormat As String = files(e.DataSourceIndex).DocumentFormat
            Select Case documentFormat
                Case ".pdf"
                    Dim largestEdgeLength As Integer = Math.Max(e.DesiredThumbnailSize.Width, e.DesiredThumbnailSize.Height)
                    e.ThumbnailImage = GenerateImageFromPDF(filePath, largestEdgeLength)
                Case ".xlsx"
                    e.ThumbnailImage = GenerateImageFromExcel(filePath)
                Case ".docx"
                    e.ThumbnailImage = GenerateImageFromWord(filePath)
            End Select
        End Sub
    End Class
End Namespace
